docs/GDPs/gdp-0001-standalone-cli.rst

```diff
@@ -0,0 +1,388 @@
+
+In support of this, two more features are proposed:
+
+ * A mechanism to allow gtscript files as if they were python modules
+ * A lazy variant or replacement of the `stencil` decorator, returning an object that supports manual stepwise compilation.
+
+Limited Scope
++++++++++++++
+
+In order to support the usecases listed below the input python files must be written in
+gtscript DSL without explicitly importing GT4Py. Instead of the decorators provided by the
+`gtscript` module, comments may be used, or functions may be inferred to be a `stencil` or `submodule`
+depending on whether they return something or not. In order to stay maintainable `gtpyc`
+will not add any new logic beyond reading input python code and command line options.
+
+However, the final version will rely on more fine grained control over when and where backends create and store intermediate source files, which should become part of the backend API to be used for run-time compilation in order to avoid redundancy and guarantee maintainability.
+
+Other language projects
++++++++++++++++++++++++
+
+In order to benefit from the higher abstraction level of the GT4Py eDSL it should
+not be required to run python code at runtime. Especially for existing programs
+written in other languages it makes more sense to link to libraries created by GT4Py
+as part of the build process of the host program.
+
+Avoiding runtime dependency
++++++++++++++++++++++++++++
+
+Even for python projects it may be desirable to distribute only the extension
+modules created by GT4Py, not the code that generated them,
+thus requiring the end user to only install GridTools, not GT4Py.
+
+Licensing
++++++++++
+
+Using only the GT4Py generated stencils in a project without depending on GT4Py at runtime
+allows to use a licence other than GPL3 in said project without the express permission of CSCS.
+
+Flexibility
++++++++++++
+
+The import mechanism will allow the flexibility to define `gtscript` objects in `gtscript` files and using them in python code
+without extra steps (as if they were defined directly in python), yet also compiling them into other language sources / bindings from the same code base just by running the CLI tool on them. This allows prototyping in python without making a final choice as to project language and license.
+
+
+Usage and Impact
+----------------
+
+Basic CLI usage
++++++++++++++++
+
+The usage is explained best using a small example.
+
+Assume the following file structure:
+
+.. code-block:: bash
+
+   $ tree .
+   pwd
+   ├── constants.py
+   └── stencils.gtpy
+
+`stencils.py` contains the `gtscript` code to be compiled to stencils. The contents might look something like the following example.
```

Comment:
Is this supposed to be `stencils.gtpy`?

Reply:
Yes, should be fixed in the latest version. Thanks